### ISSUE TITLE
Consistently Interactive

### ISSUE THREAD

**Issue by sfrieson-tm** [GoogleChrome/lighthouse#5573]
## Bug report

First off I just want to say that this issue may not actually lie with Lighthouse, and I may just be looking for advice as to what to do to get a clearer score. Thank you very much for this fantastic and helpful tool.

#### Provide the steps to reproduce
1. Run LH performance on https://www.theatermania.com/

#### What is the current behavior?
`NO_TTI_NETWORK_IDLE_PERIOD` error presents for the Consistently Interactive score, and I also wonder if it is bringing down to the First Interactive score as well. I think it might have to do with a Google Ad (DFP) pinging at a 100ms interval. Screenshots have been included to show that it seems like the app is interactive (and when I actually try to use it I can).

#### What is the expected behavior?
I expect to see a more accurate score and time.

#### Environment Information
* Affected Channels: CLI,  Extension, DevTools
* Lighthouse version: this has presented at least for the last several versions of Chrome DevTools, and a few for the CLI. I haven't been tracking the extension as much since it's in DevTools.
* Operating System: macOS 10.13, macOS 10.12, Windows 10

**Screenshots**
Chrome DevTools Performance Trace (screenshots show very little change after a while)
![image](https://user-images.githubusercontent.com/28020529/41993441-7985bbb8-7a19-11e8-8b37-64a38d498917.png)

Full Trace
![image](https://user-images.githubusercontent.com/28020529/41993510-b7761594-7a19-11e8-9f15-c8af9820fea0.png)

Long tail of trace showing repetitive pattern.
![image](https://user-images.githubusercontent.com/28020529/41993556-e3197240-7a19-11e8-80b3-896ec2822f08.png)

Detail of one segment of that pattern
![image](https://user-images.githubusercontent.com/28020529/41993615-043c594c-7a1a-11e8-96fc-f69a0f65eac1.png)

**Comment by patrickhulce** [comment]
v3 seems to have fixed the reporting issue for consistently interactive (now called "Time to Interactive"), so you should get useful numbers now.

As for improving perf... yikes that's some terrible ad behavior! Most of the scripting cost on your page is from 3rd parties, so if you're in a position to scale back reliance on some of these, then great. Otherwise, we're working on some broader initiatives to surface the cost of third party scripts and apply pressure to cleanup. Hang tight and do what you can to drive down the cost of the app.js bundle :)
![image](https://user-images.githubusercontent.com/2301202/42000074-6e205a08-7a14-11e8-92a0-854986d216fc.png)

**Comment by sfrieson-tm** [comment]
Thank you very much for looking into this. There are definitely improvements we can make in our app, but I figured I'd ask about the biggest offense first, and see if we could get more accurate timing info to compare improvements against.  I'll wait for v3 to stabilize and check back again!

**Comment by gja** [comment]
Hey @sfrieson-tm. Did you manage to fix any of these issues? We've been seeing identical behaviour, where our app loads pretty quickly, but GTM / DFP takes forever.

**Comment by sfrieson-tm** [comment]
Hey @gja, I would say that we've _fixed_ any of the issues but we have mitigated it a bit. This solution will not work for everyone, but we've delayed ad load until after content load.  Ironically enough, we recent got an insight on DFP saying that our ads load slower than other sites in our industry. 😖 Not everything can be first. It's getting the user want they want faster, we haven't yet seen a reduction in ad impressions, and no one on our ads team has complained.

**Comment by gja** [comment]
Hi @sfrieson-tm, thanks for the response! How long are you delaying your ad loads?

I believe as per the spec, TTI waits for 5 seconds of network silence, and we saw a pretty big drop in ad impressions when we tried this out.

**Comment by sfrieson-tm** [comment]
@gja I just want to reiterate that our performance metrics are still not golden, they're just a lot better.  We've done more than just delay ad loads, we've tried to delay lots of non-critical (ie below-the-fold, secondary content, secondary requests) using a mix of onLoad and requestFrameAnimation handlers.  Specifically to your question we've delayed ads with the onContentLoad handler.  We still have a red number for TTI but it's about 60% better than what it was.
